Description: Fix cros-libva VA-API struct compatibility with libva >= 2.22
 libva 2.22 added new fields (seg_id_block_size, va_reserved8) to
 VAEncPictureParameterBufferVP9 and may add fields to other structs
 in future releases. Replace explicit va_reserved field initialization
 with Rust struct update syntax (..Default::default()) to make all
 VA-API buffer type constructors forward-compatible.
Author: Greg Lamberson <greg@lamco.io>
Forwarded: https://github.com/nicira/cros-libva/issues
Last-Update: 2026-02-28
---
diff -ruN -x '*.orig' a/vendor/cros-libva/src/buffer/av1.rs b/vendor/cros-libva/src/buffer/av1.rs
--- a/vendor/cros-libva/src/buffer/av1.rs	2006-07-23 20:21:28.000000000 -0500
+++ b/vendor/cros-libva/src/buffer/av1.rs	2026-02-28 07:17:47.404502162 -0600
@@ -99,7 +99,7 @@
             segment_info_fields,
             feature_data,
             feature_mask,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -191,7 +191,7 @@
             cr_mult,
             cr_luma_mult,
             cr_offset,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -285,7 +285,7 @@
             wmtype,
             wmmat,
             invalid,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -507,7 +507,7 @@
             cdef_uv_strengths,
             loop_restoration_fields,
             wm,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -552,7 +552,7 @@
             tg_end,
             anchor_frame_idx,
             tile_idx_in_tile_list,
-            va_reserved: Default::default(),
+            ..Default::default()
         });
     }
 
@@ -915,7 +915,7 @@
             segment_number,
             feature_data,
             feature_mask,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -964,7 +964,7 @@
             wmtype,
             wmmat,
             invalid,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -1146,7 +1146,7 @@
         Self(Box::new(bindings::VAEncTileGroupBufferAV1 {
             tg_start,
             tg_end,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
diff -ruN -x '*.orig' a/vendor/cros-libva/src/buffer/h264.rs b/vendor/cros-libva/src/buffer/h264.rs
--- a/vendor/cros-libva/src/buffer/h264.rs	2006-07-23 20:21:28.000000000 -0500
+++ b/vendor/cros-libva/src/buffer/h264.rs	2026-02-28 07:17:47.404502162 -0600
@@ -24,7 +24,7 @@
             flags,
             TopFieldOrderCnt: top_field_order_cnt,
             BottomFieldOrderCnt: bottom_field_order_cnt,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -179,7 +179,7 @@
             second_chroma_qp_index_offset,
             pic_fields,
             frame_num,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -342,7 +342,7 @@
             chroma_weight_l1_flag,
             chroma_weight_l1,
             chroma_offset_l1,
-            va_reserved: Default::default(),
+            ..Default::default()
         };
 
         self.0.push(buf);
@@ -370,7 +370,7 @@
         Self(Box::new(bindings::VAIQMatrixBufferH264 {
             ScalingList4x4: scaling_list4x4,
             ScalingList8x8: scaling_list8x8,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
diff -ruN -x '*.orig' a/vendor/cros-libva/src/buffer/hevc.rs b/vendor/cros-libva/src/buffer/hevc.rs
--- a/vendor/cros-libva/src/buffer/hevc.rs	2006-07-23 20:21:28.000000000 -0500
+++ b/vendor/cros-libva/src/buffer/hevc.rs	2026-02-28 07:17:47.404502162 -0600
@@ -17,7 +17,7 @@
             picture_id,
             pic_order_cnt,
             flags,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 
@@ -253,7 +253,7 @@
             pps_tc_offset_div2,
             num_extra_slice_header_bits,
             st_rps_bits,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -555,7 +555,7 @@
             num_entry_point_offsets,
             entry_offset_to_subset_array,
             slice_data_num_emu_prevn_bytes,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -668,7 +668,7 @@
             ScalingList32x32: scaling_list32x32,
             ScalingListDC16x16: scaling_list_dc16x16,
             ScalingListDC32x32: scaling_list_dc32x32,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -861,7 +861,7 @@
             max_bytes_per_pic_denom,
             max_bits_per_min_cu_denom,
             scc_fields,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -1012,7 +1012,7 @@
             hierarchical_level_plus1,
             va_byte_reserved,
             scc_fields,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -1144,7 +1144,7 @@
             slice_fields,
             pred_weight_table_bit_offset,
             pred_weight_table_bit_length,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
diff -ruN -x '*.orig' a/vendor/cros-libva/src/buffer/mpeg2.rs b/vendor/cros-libva/src/buffer/mpeg2.rs
--- a/vendor/cros-libva/src/buffer/mpeg2.rs	2006-07-23 20:21:28.000000000 -0500
+++ b/vendor/cros-libva/src/buffer/mpeg2.rs	2026-02-28 07:17:47.408502184 -0600
@@ -79,7 +79,7 @@
             picture_coding_type,
             f_code,
             picture_coding_extension,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -118,7 +118,7 @@
             slice_vertical_position,
             quantiser_scale_code,
             intra_slice_flag,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -157,7 +157,7 @@
             non_intra_quantiser_matrix,
             chroma_intra_quantiser_matrix,
             chroma_non_intra_quantiser_matrix,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
diff -ruN -x '*.orig' a/vendor/cros-libva/src/buffer/vp8.rs b/vendor/cros-libva/src/buffer/vp8.rs
--- a/vendor/cros-libva/src/buffer/vp8.rs	2006-07-23 20:21:28.000000000 -0500
+++ b/vendor/cros-libva/src/buffer/vp8.rs	2026-02-28 07:17:47.408502184 -0600
@@ -122,7 +122,7 @@
             uv_mode_probs,
             mv_probs,
             bool_coder_ctx,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -156,7 +156,7 @@
             macroblock_offset,
             num_of_partitions,
             partition_size,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -178,7 +178,7 @@
     pub fn new(quantization_index: [[u16; 6usize]; 4usize]) -> Self {
         Self(Box::new(bindings::VAIQMatrixBufferVP8 {
             quantization_index,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -200,7 +200,7 @@
     pub fn new(dct_coeff_probs: [[[[u8; 11usize]; 3usize]; 8usize]; 4usize]) -> Self {
         Self(Box::new(bindings::VAProbabilityDataBufferVP8 {
             dct_coeff_probs,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -243,7 +243,7 @@
             bits_per_second,
             intra_period,
             reference_frames,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -387,7 +387,7 @@
             sharpness_level,
             clamp_qindex_high,
             clamp_qindex_low,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
diff -ruN -x '*.orig' a/vendor/cros-libva/src/buffer/vp9.rs b/vendor/cros-libva/src/buffer/vp9.rs
--- a/vendor/cros-libva/src/buffer/vp9.rs	2026-02-27 22:32:53.000000000 -0600
+++ b/vendor/cros-libva/src/buffer/vp9.rs	2026-02-28 07:17:47.404502162 -0600
@@ -115,7 +115,6 @@
             segment_pred_probs,
             profile,
             bit_depth,
-            va_reserved: Default::default(),
             ..Default::default()
         }))
     }
@@ -184,7 +183,7 @@
             luma_dc_quant_scale,
             chroma_ac_quant_scale,
             chroma_dc_quant_scale,
-            va_reserved: Default::default(),
+            ..Default::default()
         })
     }
 }
@@ -207,7 +206,7 @@
             slice_data_offset,
             slice_data_flag,
             seg_param,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -237,7 +236,7 @@
             long_term_indication,
             next_frame_width,
             next_frame_height,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 }
@@ -262,7 +261,7 @@
             kf_max_dist,
             bits_per_second,
             intra_period,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
@@ -435,7 +434,7 @@
             skip_frame_flag,
             number_skip_frames,
             skip_frames_size,
-            va_reserved: Default::default(),
+            ..Default::default()
         }))
     }
 
